### ISSUE TITLE
Kafka testcontainers

### ISSUE THREAD

**Issue by janessbach** [testcontainers/testcontainers-scala#47]
not really an issue, just wanted to provide a simple Kafka test container

```
package testing.testcontainers

import com.dimafeng.testcontainers.GenericContainer
import org.testcontainers.containers.{ Network, SocatContainer }
import org.testcontainers.utility.Base58

class KafkaContainer(version: String) extends GenericContainer(
  dockerImage = s"confluentinc/cp-kafka:$version",
  env = KafkaContainer.defaultEnv(KafkaContainer.KAFKA_PORT)) {

  import KafkaContainer._

  val kafkaNetwork: Network = Network.newNetwork

  var proxy: SocatContainer = new SocatContainer()
    .withNetwork(kafkaNetwork)
    .withTarget(KAFKA_PORT, networkAliases.head)
    .withTarget(ZOOKEEPER_PORT, networkAliases.head)

  proxy.start()

  configure(containerSettings => {

    containerSettings.withNetwork(kafkaNetwork)
    containerSettings.withNetworkAliases("kafka-" + Base58.randomString(6))
    containerSettings.withExposedPorts(KAFKA_PORT)

    containerSettings.addExposedPort(ZOOKEEPER_PORT)
    containerSettings.withEnv("KAFKA_ZOOKEEPER_CONNECT", "localhost:2181")

    containerSettings.withCommand("sh", "-c",
      "printf 'clientPort=2181\ndataDir=/var/lib/zookeeper/data\ndataLogDir=/var/lib/zookeeper/log' > /zookeeper.properties"
        + " && zookeeper-server-start /zookeeper.properties"
        + " & /etc/confluent/docker/run")

    containerSettings.withEnv("KAFKA_ADVERTISED_LISTENERS", "BROKER://" + networkAliases.head + ":9092" + "," + getBootstrapServers)

  })

  def getBootstrapServers: String =
    String.format("PLAINTEXT://%s:%s", proxy.getContainerIpAddress, proxy.getFirstMappedPort)

}

object KafkaContainer {
  val KAFKA_PORT = 9093
  val ZOOKEEPER_PORT = 2181

  def apply(version: String = "3.2.1"): KafkaContainer = new KafkaContainer(version)

  def defaultEnv(kafkaPort: Int): Map[String, String] = Map(
    "KAFKA_LISTENERS" -> s"PLAINTEXT://0.0.0.0:$kafkaPort,BROKER://0.0.0.0:9092",
    "KAFKA_LISTENER_SECURITY_PROTOCOL_MAP" -> "BROKER:PLAINTEXT,PLAINTEXT:PLAINTEXT",
    "KAFKA_INTER_BROKER_LISTENER_NAME" -> "BROKER",
    "KAFKA_BROKER_ID" -> "1",
    "KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR" -> "1",
    "KAFKA_OFFSETS_TOPIC_NUM_PARTITIONS" -> "1",
    "KAFKA_LOG_FLUSH_INTERVAL_MESSAGES" -> Long.MaxValue.toString,
    "KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS" -> "0")
}
```

**Comment by bsideup** [comment]
There is KafkaContainer in the upstream Java version of the library, why would you re-implement it in Scala? Just curious :)

**Comment by dimafeng** [comment]
@janessbach thank you for the contribution! Do you have a simple test case\spec for this container?

@bsideup I'll check how it's different from upstream version. Anyways it's a good start, having a test case will help with potentially moving to the upstream version (if needed).

**Comment by mbuccini** [comment]
It works like a charm. Thanks a lot for sharing!

**Comment by dimafeng** [comment]
Implemented #60
